Kudu.Core/Environment.cs

```diff
@@ -95,7 +96,17 @@ public class Environment : IEnvironment
             _siteExtensionSettingsPath = Path.Combine(SiteRootPath, Constants.SiteExtensionsCachePath);
             _diagnosticsPath = Path.Combine(SiteRootPath, Constants.DiagnosticsPath);
             _locksPath = Path.Combine(SiteRootPath, Constants.LocksPath);
-            _sshKeyPath = Path.Combine(rootPath, Constants.SSHKeyPath);
+
+            string siteName = System.Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME");
+            if (OSDetecter.IsOnWindows() || rootPath.TrimEnd(new char[] { '/', '\\' }).EndsWith(siteName, StringComparison.OrdinalIgnoreCase))
```

Comment:
I do agree that it's worth making it runnable both on normal Linux and in container, to help testing.